File: plug-in/PDFDuplicateFile/C#/ChooseBackupFolder.cs
using System;
using System.Windows.Forms;
using Microsoft.VisualBasic;

namespace PDFChooseBackupFolder
{
    public partial class ChooseBackupFolder : Form
    {
        public ChooseBackupFolder()
        {
            InitializeComponent();
        }

        private void CenterForm(Form frm, Form parent = null/* TODO Change to default(_) if this is not a reference type */)
        {
            System.Drawing.Rectangle r;
            if (parent != null)
                r = parent.RectangleToScreen(parent.ClientRectangle);
            else
                r = Screen.FromPoint(frm.Location).WorkingArea;

            var x = r.Left + (r.Width - frm.Width) / 2;
            var y = r.Top + (r.Height - frm.Height) / 2;
            frm.Location = new System.Drawing.Point(x, y);
        }

        private void btnBrowseFolder_Click(object sender, EventArgs e)
        {
            FolderBrowserDialog folderBrowserDialog1 = new FolderBrowserDialog();
            if (folderBrowserDialog1.ShowDialog() == DialogResult.OK)
            {
                txtBackupFolder.Text = folderBrowserDialog1.SelectedPath;
            }
        }

        private void btnCancel_Click(object sender, EventArgs e)
        {
            this.Close();
        }

        private void btnOK_Click(object sender, EventArgs e)
        {
            Interaction.SaveSetting(PDFDuplicateFile.WIN2PDF_COMPANY, PDFDuplicateFile.WIN2PDF_PRODUCT, PDFDuplicateFile.DUPFOLDER_SETTING, txtBackupFolder.Text);
            this.Close();
        }

        private void ChooseBackupFolder_Load(object sender, EventArgs e)
        {
            txtBackupFolder.Text = Interaction.GetSetting(PDFDuplicateFile.WIN2PDF_COMPANY, PDFDuplicateFile.WIN2PDF_PRODUCT, PDFDuplicateFile.DUPFOLDER_SETTING, Environment.SpecialFolder.MyDocuments + @"\backup");
            CenterForm(this);
        }
    }
}


//namespace PDFDuplicateFile
//{
//    public partial class ChooseBackupFolder : Form
//    {
//        public ChooseBackupFolder()
//        {
//            InitializeComponent();
//        }
//    }
//}

File: plug-in/PDFDuplicateFile/VB.NET/ChooseBackupFolder.vb
Imports System.Windows.Forms
Imports Microsoft.WindowsAPICodePack.Dialogs

Public Class ChooseBackupFolder
    Private Sub btnBrowseFolder_Click(sender As Object, e As EventArgs) Handles btnBrowseFolder.Click
        ' Open a folder picker dialog to allow the user to select a folder
        Dim dialog As CommonOpenFileDialog = New CommonOpenFileDialog()
        dialog.IsFolderPicker = True ' Ensure the dialog only allows folder selection
        If dialog.ShowDialog() = CommonFileDialogResult.Ok Then
            ' Set the selected folder path to the text box
            txtBackupFolder.Text = dialog.FileName
        End If
    End Sub

    Private Sub btnOK_Click(sender As Object, e As EventArgs) Handles btnOK.Click
        SaveSetting(PDFDuplicateFile.WIN2PDF_COMPANY, PDFDuplicateFile.WIN2PDF_PRODUCT, PDFDuplicateFile.DUPFOLDER_SETTING, txtBackupFolder.Text)
        Me.Close()
    End Sub

    Private Sub btnCancel_Click(sender As Object, e As EventArgs) Handles btnCancel.Click
        Me.Close()
    End Sub

    Private Sub CenterForm(ByVal frm As Form, Optional ByVal parent As Form = Nothing)
        Dim r As Drawing.Rectangle
        If parent IsNot Nothing Then
            r = parent.RectangleToScreen(parent.ClientRectangle)
        Else
            r = Screen.FromPoint(frm.Location).WorkingArea
        End If

        Dim x = r.Left + (r.Width - frm.Width) \ 2
        Dim y = r.Top + (r.Height - frm.Height) \ 2
        frm.Location = New Drawing.Point(x, y)
    End Sub


    Private Sub ChooseBackupFolder_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        txtBackupFolder.Text = Interaction.GetSetting(WIN2PDF_COMPANY, WIN2PDF_PRODUCT, DUPFOLDER_SETTING, My.Computer.FileSystem.SpecialDirectories.MyDocuments + "\backup")
        CenterForm(Me)
    End Sub
End Class